src/test/java/com/exasol/adapter/dialects/elasticsearch/ElasticSearchGateway.java

```diff
@@ -1,83 +1,143 @@
+                    ElasticsearchContainer.ELASTICSEARCH_DEFAULT_PASSWORD);
+        } else {
+            return ElasticSearchGateway.connect("http://" + container.getHttpHostAddress(), "elastic",
+                    ElasticsearchContainer.ELASTICSEARCH_DEFAULT_PASSWORD);
+        }
+    }
+
+    private static ElasticSearchGateway connectViaTls(final String httpHostAddress, final SSLContext sslcontext,
+            final String username, final String password) {
+        final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username, password));
+
+        final HttpHost host = HttpHost.create(httpHostAddress);
+        final HostnameVerifier hostnameVerifier = new TrustSpecificHostnameVerifier(host.getHostName());
+        final RestClient restClient = RestClient.builder(host)
+                .setHttpClientConfigCallback(
+                        httpClientBuilder -> httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider)
+                                .setSSLHostnameVerifier(hostnameVerifier).setSSLContext(sslcontext))
+                .build();
+        final ElasticsearchTransport transport = new RestClientTransport(restClient, new JsonbJsonpMapper());
+        final ElasticsearchClient client = new ElasticsearchClient(transport);
+        return new ElasticSearchGateway(client);
+    }
+
+    private static ElasticSearchGateway connect(final String httpHostAddress, final String username,
+            final String password) {
+        final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username, password));
+
+        final HttpHost host = HttpHost.create(httpHostAddress);
+        final RestClient restClient = RestClient.builder(host).build();
+        final ElasticsearchTransport transport = new RestClientTransport(restClient, new JsonbJsonpMapper());
+        final ElasticsearchClient client = new ElasticsearchClient(transport);
+        return new ElasticSearchGateway(client);
     }
 
     /**
      * Index a new document in Json format in the data source.
-     *
-     * @param indexName
-     * @param jsonSource
-     * @throws IOException
      */
-    public void indexDocument(final String indexName, final String jsonSource) throws IOException {
-        final IndexRequest indexRequest = new IndexRequest(indexName).source(jsonSource, XContentType.JSON);
-        this.client.index(indexRequest, RequestOptions.DEFAULT);
+    public void indexDocument(final String indexName, final String jsonSource) {
+        try {
+            client.index(i -> i.index(indexName).withJson(new StringReader(jsonSource)));
+        } catch (final IOException exception) {
+            throw new UncheckedIOException("Failed to index document", exception);
+        }
```

Comment:
Would it make sense to have a central method? 
```java
    public void createIndex(final String indexName) {
        withUncheckedIOException("Failed to create index", //
                () -> this.client.indices().create(i -> i.index(indexName)));
    }

    @FunctionalInterface
    interface RunnableWithIOException {
        void run() throws IOException;
    }

    private void withUncheckedIOException(final String exceptionMessage, final RunnableWithIOException runnable) {
        try {
            runnable.run();
        } catch (final IOException exception) {
            throw new UncheckedIOException(exceptionMessage, exception);
        }
    }

```

Reply:
I understand. But I think this is overkill for a class just used in tests.